seacrowd/sea_datasets/coco_35l/coco_35l.py

```diff
@@ -0,0 +1,225 @@
+    "trans_dev": "https://storage.googleapis.com/crossmodal-3600/coco_mt_dev.jsonl.gz",
+}
+
+_SUPPORTED_TASKS = [Tasks.IMAGE_CAPTIONING]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+_LANGS = ["fil", "id", "th", "vi"]
+
+
+class Coco35LDataset(datasets.GeneratorBasedBuilder):
+    """
+    COCO-35L is a machine-generated image caption dataset, constructed by translating COCO Captions (Chen et al., 2015) to the other 34 languages using Google’s machine translation API.
+    """
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [SEACrowdConfig(name=f"{_DATASETNAME}_{lang}_source", version=datasets.Version(_SOURCE_VERSION), description=f"{_DATASETNAME}_{lang} source schema", schema="source", subset_id=f"{_DATASETNAME}_{lang}",) for lang in _LANGS] + [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{lang}_seacrowd_imtext",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME}_{lang} SEACrowd schema",
+            schema="seacrowd_imtext",
+            subset_id=f"{_DATASETNAME}_{lang}",
+        )
+        for lang in _LANGS
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_{sorted(_LANGS)[0]}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "image_paths": datasets.Value("string"),
+                    "src_lang": datasets.Value("string"),
+                    "caption_tokenized": datasets.Value("string"),
+                    "trg_lang": datasets.Value("string"),
+                    "translation_tokenized": datasets.Value("string"),
+                    "backtranslation_tokenized": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "seacrowd_imtext":
+            features = schemas.image_text_features()
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        trans_train_path = dl_manager.download_and_extract(_URLS["trans_train"])
+        trans_val_path = dl_manager.download_and_extract(_URLS["trans_dev"])
+
+        coco2014_train_val_annots_path = dl_manager.download_and_extract(_URLS["coco2014_train_val_annots"])
+        coco2014_val_images_path = dl_manager.download_and_extract(_URLS["coco2014_val_images"])
+
+        # coco2017_train_val_annots_path = dl_manager.download_and_extract(_URLS["coco2017_train_val_annots"])
+        # coco2017_train_images_path = dl_manager.download_and_extract(_URLS["coco2017_train_images"])
```

Comment:
Used to test the loader since I don't know which COCO version to be used. 2014 has more missing ```image_id```s compared to 2017, but the captions don't match. Thus it should be 2014. Will remove if everything is ok.